backend/src/main/java/org/sefglobal/core/academix/service/ItemService.java

```diff
@@ -0,0 +1,30 @@
+package org.sefglobal.core.academix.service;
+
+import org.sefglobal.core.exception.BadRequestException;
+import org.sefglobal.core.academix.model.SubCategory;
+import org.sefglobal.core.academix.projections.CustomItem;
+import org.sefglobal.core.academix.repository.ItemRepository;
+import org.sefglobal.core.academix.repository.SubCategoryRepository;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class ItemService {
+    private final SubCategoryRepository subCategoryRepository;
+    private final ItemRepository itemRepository;
+
+    public ItemService(SubCategoryRepository subCategoryRepository, ItemRepository itemRepository) {
+        this.subCategoryRepository = subCategoryRepository;
+        this.itemRepository = itemRepository;
+    }
+
+    public List<CustomItem> getAllItemsBySubCategory(Long subCategoryId, int pageNumber, int pageSize) throws BadRequestException {
+        Pageable pageable = PageRequest.of(pageNumber, pageSize);
+        SubCategory subCategory = subCategoryRepository.findById(subCategoryId)
+                .orElseThrow(() -> new BadRequestException("Error, Sub Category by id:"+subCategoryId+" doesn't exist."));
+        return itemRepository.getAllBySubCategories(subCategory, pageable);
+    }
```

Comment:
Hi, is there any reason for handling this method and `getAllSubCategoryByCategory ` method in `SubCategoryService ` in two different approaches?